x/interchainstaking/keeper/grpc_query.go

```diff
@@ -193,6 +193,35 @@ func (k *Keeper) Receipts(c context.Context, req *types.QueryReceiptsRequest) (*
 	return &types.QueryReceiptsResponse{Receipts: receipts}, nil
 }
 
+func (k *Keeper) TxStatus(c context.Context, req *types.QueryTxStatusRequest) (*types.QueryTxStatusResponse, error) {
+	// TODO: implement pagination
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	if req.GetTxHash() == "" {
+		return nil, status.Error(codes.InvalidArgument, "tx hash cannot be empty")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	zone, found := k.GetZone(ctx, req.GetChainId())
+	if !found {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("no zone found matching %s", req.GetChainId()))
+	}
+
+	txReceipt := &types.Receipt{}
+
+	k.IterateZoneReceipts(ctx, &zone, func(_ int64, receipt types.Receipt) (stop bool) {
```

Comment:
No need to iterate the zones here:

`receipt, found := k.GetReceipt(types.GetReceiptKey(req.GetChainId(), req.GetTxHash()))`

Reply:
Strictly speaking we don't actually need to validate the zone, just the getReceipt would suffice - I guess the only reason to validate the zone is because is to tell the user they are looking for a chain that is not onboarded (e.g. a typo), but given that this endpoint is primarily used by the F/E I'm less inclined to care for this scenario.

Reply:
added the changes and removed zone validation, returning receipt not found error in case of mismatch